src/UI/Utils/useIsDirty.test.tsx

```diff
@@ -0,0 +1,175 @@
+import React, { useState } from "react";
+import { act, cleanup, render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { Field, ServiceInstance } from "@/Test";
+import useIsDirty from "./useIsDirty";
+
+// Test case setup checks indirectly boolean value by conditional rendering of a span tag
+const setup = (baseObject, testCompareObject, newValue, overrideValue?) => {
+  const Component: React.FC = () => {
+    const [testObject, setTestObject] = useState(baseObject);
+    const { isDirty, overrideState } = useIsDirty(
+      testObject,
+      testCompareObject
+    );
+    return (
+      <div>
+        <button onClick={() => setTestObject(newValue)}>Change Data</button>
+        <button onClick={() => overrideState(overrideValue)}>
+          override Data
+        </button>
+        {isDirty && <span role="isDirtyValue">true</span>}
+      </div>
+    );
+  };
+  const component = <Component />;
+  return { component };
+};
+
+afterEach(() => {
+  cleanup;
+});
+test("GIVEN useIsDirty WHEN hook's parameter that is being checked isn't changed from it's default THEN isDirty value is falsy and span isn't visible in any moment of lifecycle", async () => {
+  Field.nestedEditable, ServiceInstance.nestedEditable.candidate_attributes;
+  const { component } = setup(
+    {
+      property: "",
```

Comment:
I extracted objects to variables, to avoid repetition, I don't know why I didn't do it in the first place, but if it comes what they mean it is told when you hover over setup()

Reply:
I can't hover when reviewing :) And when functions get complicated, hovering isn't always clear as to what is being passed :)